### PR TITLE
[doc] Add basic documentation for host software

### PR DESCRIPTION
**Pull request by jwnrt**
This PR adds just the minimum documentation for OpenTitan's host software.

This will need expanding in the future, but hopefully adding these files will prompt people to add to them and keep them up to date in the future.

Future work:

* If anybody can provide more detailed documentation here, that would be greatly appreciated.
* I couldn't work out exactly what the `cryptotest` library was so have left that out for now.
* I couldn't work out what the `tpm2_test_server` was used for, but have copied its `about` line from the code.

### CODE REVIEW

File: sw/host/opentitantool/README.md
Comment (pamaury) [inline sw/host/opentitantool/README.md:28]: Missing "r"?

**Review by engdoreis — APPROVED** [review]
Thanks @jwnrt for the initiative, let's wait for more reviews before merging this PR.
I would also open an issue to keep track of work that need to be done.

**Comment by jadephilipoom** [comment]
> * I couldn't work out exactly what the `cryptotest` library was so have left that out for now.

It was the start of a library for running randomized/large test sets on the cryptolib, but the effort got abandoned over a year ago and it's currently unused. We can safely ignore it in the docs for now, or even delete it from the repo and resurrect it from git history later if needed.

File: sw/host/hsmtool/README.md
Comment (HU90m) [inline sw/host/hsmtool/README.md:3]:
```suggestion
The OpenTitan Hardware Security Module (HSM) tool can be used to generate and manage keys used for OpenTitan signing ceremonies.
```

Do we have a glossary of terms? If so maybe HSM should go in there; it's pretty general.

File: sw/host/opentitanlib/README.md
Comment (HU90m) [inline sw/host/opentitanlib/README.md:5]:
```suggestion
OpenTitanLib is used by OpenTitanTool and by some tests which need finer grained control over the device under test.
```

**Review by HU90m — APPROVED** [review]
Brilliant, thanks James!

Do we have a glossary of terms? If not, should we make one?

File: sw/host/hsmtool/README.md
Comment (jwnrt) [reply sw/host/hsmtool/README.md:3]: Yeah, OpenTitan definitely needs a glossary, but I'll open an issue for it rather than make one in this PR

Comment (jwnrt) [reply sw/host/hsmtool/README.md]: Opened an issue for the glossary here: https://github.com/lowRISC/opentitan/issues/19437

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**
